analyzers/its/Directory.Build.targets

```diff
@@ -56,10 +58,10 @@
 
     <PropertyGroup>
       <!-- Set the path to the SARIF report -->
-      <ErrorLog Condition="'$(TargetFramework)' != ''">$(AnalysisOutputPath)-$(ProjectGuid)-$(TargetFramework).json</ErrorLog>
-      <ErrorLog Condition="'$(TargetFramework)' == ''">$(AnalysisOutputPath)-$(ProjectGuid).json</ErrorLog>
-      <RazorCompilationErrorLog Condition="'$(TargetFramework)' != ''">$(AnalysisOutputPath)-$(ProjectGuid)-$(TargetFramework).Views.json</RazorCompilationErrorLog>
-      <RazorCompilationErrorLog Condition="'$(TargetFramework)' == ''">$(AnalysisOutputPath)-$(ProjectGuid).Views.json</RazorCompilationErrorLog>
+      <ErrorLog Condition="'$(TargetFramework)' != ''">$(DiagnosticsPathWithTFM).json</ErrorLog>
```

Comment:
Precomputing this into a variable is a good idea, but it should be done the other way around. There should be a single variable, that will be computed either with one condition or other. And here, it should just use the value. So it would go something like
```
    <ErrorLogPrefix Condition=...==....>....
    <ErrorLogPrefix Condition=....!=....>....
...
    <ErrorLog>$(ErrorLogPrefix).json...
    <RazorCompilationErrorLog>$(ErrorLogPrefix).Views.json...
```